e2etest/pkg/frr/config/config.go

```diff
@@ -36,24 +37,33 @@ router bgp {{.ASN}}
   neighbor {{.Addr}} bfd
 {{- end -}}
 {{- end }}
-  address-family {{.IPFamily}} unicast
-{{range .Neighbors }}
+{{- if ne (len .V4Neighbors) 0}}
+  address-family ipv4 unicast
+{{range .V4Neighbors }}
     neighbor {{.Addr}} next-hop-self
     neighbor {{.Addr}} activate
-    {{ if eq .IPFamily "ipv6" -}}
-    neighbor {{.Addr}} route-map RMAP in
-    {{- end }}
 {{- end }}
   exit-address-family
+{{- end }}
+{{- if ne (len .V6Neighbors) 0}}
+  address-family ipv6 unicast
+{{range .V6Neighbors }}
+    neighbor {{.Addr}} next-hop-self
+    neighbor {{.Addr}} activate
+    neighbor {{.Addr}} route-map RMAP in
+{{- end }}
+exit-address-family
+{{- end }}
 
 `
 
 type RouterConfig struct {
-	ASN       uint32
-	Neighbors []*NeighborConfig
-	BGPPort   uint16
-	Password  string
-	IPFamily  string
+	ASN         uint32
+	Neighbors   []*NeighborConfig
```

Comment:
we don't need NeighborConfig anymore ?

Reply:
We do for line 31
https://github.com/metallb/metallb/pull/1079/files#diff-c0cf4503bd002aff51e2073c4bca0f2c8a9c8a4d7749fd8de31aa9b23b619ed0R31

Merging the two other slices in the template seemed more hacky